src/main/kotlin/uk/gov/justice/digital/hmpps/supportadditionalneedsapi/reporting/repository/PrisonActivitySummaryRepository.kt

```diff
@@ -0,0 +1,142 @@
+               GROUP BY created_at_prison
+           )
+      SELECT COALESCE(
+                     aln.created_at_prison,
+                     elsp.created_at_prison,
+                     ch.created_at_prison,
+                     cond.created_at_prison,
+                     str.created_at_prison,
+                     supp.created_at_prison
+             ) AS created_at_prison,
+             COALESCE(aln.aln_screener, 0) AS aln_screener,
+             COALESCE(elsp.elsp_plan, 0) AS elsp_plan,
+             COALESCE(ch.challenge, 0) AS challenge,
+             COALESCE(cond.condition, 0) AS condition,
+             COALESCE(str.strength, 0) AS strength,
+             COALESCE(supp.support_strategy, 0) AS support_strategy
+      FROM aln
+               FULL OUTER JOIN elsp ON aln.created_at_prison = elsp.created_at_prison
+               FULL OUTER JOIN ch ON COALESCE(aln.created_at_prison, elsp.created_at_prison) = ch.created_at_prison
+               FULL OUTER JOIN cond ON COALESCE(
+                                               aln.created_at_prison,
+                                               elsp.created_at_prison,
+                                               ch.created_at_prison
+                                       ) = cond.created_at_prison
+               FULL OUTER JOIN str ON COALESCE(
+                                              aln.created_at_prison,
+                                              elsp.created_at_prison,
+                                              ch.created_at_prison,
+                                              cond.created_at_prison
+                                      ) = str.created_at_prison
+               FULL OUTER JOIN supp ON COALESCE(
+                                               aln.created_at_prison,
+                                               elsp.created_at_prison,
+                                               ch.created_at_prison,
+                                               cond.created_at_prison,
+                                               str.created_at_prison
+                                       ) = supp.created_at_prison
+      WHERE COALESCE(aln.created_at_prison, elsp.created_at_prison, ch.created_at_prison, cond.created_at_prison, str.created_at_prison, supp.created_at_prison) IS NOT NULL
+        AND COALESCE(
+                     aln.created_at_prison,
+                     elsp.created_at_prison,
+                     ch.created_at_prison,
+                     cond.created_at_prison,
+                     str.created_at_prison,
+                     supp.created_at_prison
+             ) NOT IN (
+                                      'ACI', 'DGI', 'FWI', 'FBI', 'PBI', 'RHI', 'WNI', 'BWI', 'BZI', 'PRI',
+                                      'UKI', 'WYI', 'ASI', 'PFI', 'UPI', 'SWI', 'PYI', 'CFI', 'CKI', 'FYI',
+                                      'FEI', 'LGI', 'MKI'
+          )
+      ORDER BY created_at_prison asc
```

Comment:
Blimey! What a query! 🤣 
This is exactly why we are using the read-replica for all these reporting queries 👍 😁